activerecord/lib/active_record/query_cache.rb

```diff
@@ -24,16 +24,19 @@ def uncached(&block)
     end
 
     def self.run
-      caching_pool = ActiveRecord::Base.connection_pool
-      caching_was_enabled = caching_pool.query_cache_enabled
+      ActiveRecord::Base.connection_handler.connection_pool_list.map do |pool|
+        caching_was_enabled = pool.query_cache_enabled
 
-      caching_pool.enable_query_cache!
+        pool.enable_query_cache!
 
-      [caching_pool, caching_was_enabled]
+        [pool, caching_was_enabled]
+      end
```

Comment:
You can reduce the number of generated arrays by passing only the pools where caching was enabled into the `complete` hook:

``` ruby
    def self.run
      pools = ActiveRecord::Base.connection_handler.connection_pool_list.select do |pool|
        unless pool.query_cache_enabled
           p.enable_query_cache!
        end
     end
    end

    def self.complete(pools)
      pools.each {|pool| pool.disable_query_cache! }
      ...
    end
```

Reply:
You're right! That approach makes the code a bit clearer too.

As you already wrote the patch... do you want to open a pull request? 😄